docs/src/developer/gpu_computations.md

```diff
@@ -88,3 +88,62 @@ performance. For example, iterating through the columns of a matrix to compute t
 norms is not efficient, as a new kernel is launched for every column. Instead, it
 is better to build the vector containing these norms, as it is a vectorized
 operation and will be much faster on the GPU.
+
+## Anti-patterns
+
+- Avoid hard coding `Array` in fields of a struct, as it prevents the use of GPU. In fact, this results in the underlying data being automatically moved to the CPU, which might not be ideal.
+- Avoid using `zeros`, `ones` as they simply the output would be an `Array` that is on CPU. Instead, use `zeros_like` constructor, or `similar`.
+
+For example,
+
+```julia
+rho_temp = zeros(T, basis.fft_size...)
+```
+
+should be written as:
+
+```julia
+rho_temp = zeros_like(basis.G_vectors, T, basis.fft_size...)
+```
+
+or 
+
+```julia
+rho_temp = similar(basis.G_vectors, T, basis.fft_size...)
+fill!(rho_temp, 0)
+```
+
+
+## Debugging
+
+First, make sure the scalar indexing is disabled as these calculations are not computed on GPU but instead on the CPU.
```

Comment:
I'd rather write `... as otherwise fallback routines on the CPU are allowed. While these are helpful for interactive debugging, these always imply a synchronisation on the device and a data transfer to main memory. In practice runs this then results in an unnecessary overhead and the calculation appearing to be struck.`

Point is that computation is not necessarily happening on the CPU (depends on what you do with the scalar data), but the sync+transfer can be a bad idea.